### DISCUSSION TITLE
Dynamic server usage: headers error with getServerSession(authOptions)

### DISCUSSION THREAD

**Discussion by saver711** [nextauthjs/next-auth#7868] [Help]
### Environment

"dependencies": {
    "@apollo/client": "^3.8.0-beta.3",
    "@apollo/experimental-nextjs-app-support": "^0.2.1",
    "@types/node": "20.3.1",
    "@types/react": "18.2.13",
    "@types/react-dom": "18.2.6",
    "autoprefixer": "10.4.14",
    "eslint": "8.43.0",
    "eslint-config-next": "13.4.6",
    "graphql": "^16.6.0",
    "next": "^13.4.7",
    "next-auth": "^4.22.1",
    "postcss": "8.4.24",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "tailwindcss": "3.3.2",
    "typescript": "5.1.3"
  },
  "devDependencies": {
    "@tailwindcss/aspect-ratio": "^0.4.2",
    "tailwindcss-rtl": "^0.9.0"
  }

### Reproduction URL

https://github.com/saver711/nextjs13-starpi

### Describe the issue

When i am logged in and try to go from http://localhost:3000/articles to => http://localhost:3000/articles/anyID
i get this error
![image](https://github.com/nextauthjs/next-auth/assets/28202010/56dbb665-1c91-44b7-93a1-8cf32204455e)

because i have getServerSession(authOptions) in my Navbar component

### How to reproduce

just 
yarn
yarn dev
in both next-front and next-back directories

### Expected behavior

to go to http://localhost:3000/articles/anyID without this error and to have session data

**Comment by suryansh-wattoai** [comment]
Hey, I am also facing the exact same issue. My code working fine in development but breaks in production with the above error.

**Comment by beyourahi** [comment]
I'm facing the same issue as well. Any solutions to this?

**Comment by gullerg** [comment]
I'm facing the same issue. Moreover, the error does not consistently appear in dev, making it even harder to reproduce. It does not appear once built/in prod.

**Comment by MrGlox** [comment]
Hi ! I just had the same problem, it seems resolved after adding getServerSession to the layout.tsx at the root of the app folder.

**Reply by gullerg** [reply]
Could you include an example of how to add it to the layout.tsx? :)

**Reply by MrGlox** [reply]
Sorry, after restart, it just broke with : import { getServerSession } from "next-auth";

But then with updating to : import { getServerSession } from "next-auth/next"; everything seems fine, I have a custom middleware for language and for authentication.

It looks like :
```
import { getServerSession } from "next-auth/next";

export default async function RootLayout({ children }: Props) {
  const session = await getServerSession(authOptions);
  // const lang = cookies()?._parsed.get("i18next").value;

  return (
    <html lang={lang} dir={dir(lang)}>
      <head />
      <body className="font-sans">
        <Providers session={session}>{children}</Providers>
      </body>
    </html>
  );
}
```

**Reply by Overclock0** [reply]
Could you show Providers file more detail when you send props session to that

**Reply by MrGlox** [reply]
Here, I use react-toastify, and react-query too, in next13 app folder we have to make a "use client" file for all contexts API

```
"use client";

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { SessionProvider } from "next-auth/react";
import { ToastContainer } from "react-toastify";

const queryClient = new QueryClient();

export function Providers({
  session,
  children,
}: {
  session: any;
  children: React.ReactNode;
}) {
  return (
    <SessionProvider session={session}>
      <QueryClientProvider client={queryClient}>
        {children}
        <ToastContainer
          position="top-right"
          autoClose={5000}
          pauseOnHover
          theme="light"
        />
      </QueryClientProvider>
    </SessionProvider>
  );
}
```

**Reply by limeojin363** [reply]
Sorry but, this doesn't work for me. In RootLayout getServerSession occurs the same error.

**Reply by limeojin363** [reply]
Oh, after updating next.js to latest version, it works well!

**Reply by gullerg** [reply]
which version is this?
